File: angularws/emsngdemo/src/app/model/employee.ts
//https://codebunk.com/b/7481100535422/
export interface Employee{
    eid:number,
    ename:string,
    email:string,
    phone:string,
    password?:string,
    address:{
        city?:string,
        country:string,
        zipcode?:number
    }
}

// interface Employee{
//     name:string,
//     city?:string
// }

// function display(data:Employee)
// {
//     console.log(data.name)
// }
// display({name:''})

File: SpringBootEMSRestDemo/src/main/java/com/boot/demo/entity/Employee.java
package com.boot.demo.entity;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import org.springframework.context.annotation.Description;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Employee {

	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@NotNull
	private int eid;
	private String ename;
	private String email;
	private String phone;
	@Size(min = 8, max=15)
	private String password;
	
	@OneToOne(mappedBy = "employee",
			cascade = {CascadeType.REMOVE})
	private Address address;
}
